onnxruntime/core/session/inference_session.cc

```diff
@@ -1131,30 +1131,34 @@ common::Status InferenceSession::Initialize() {
     // Register 2nd registries into KernelRegistryManager.
     ORT_RETURN_IF_ERROR_SESSIONID_(kernel_registry_manager_.RegisterKernels(execution_providers_));
 
+    bool loading_ort_format = !ort_format_model_bytes_.empty();
 #if !defined(ORT_MINIMAL_BUILD)
-    // add predefined transformers
-    AddPredefinedTransformers(graph_transformation_mgr_, session_options_.graph_optimization_level,
-                              transformers_to_enable_);
-
-    // apply any transformations to the main graph and any subgraphs
-    ORT_RETURN_IF_ERROR_SESSIONID_(TransformGraph(graph, graph_transformation_mgr_,
-                                                  execution_providers_, kernel_registry_manager_,
-                                                  insert_cast_transformer_,
-                                                  *session_state_));
-
-    // now that all the transforms are done, call Resolve on the main graph. this will recurse into the subgraphs.
-    ORT_RETURN_IF_ERROR_SESSIONID_(graph.Resolve());
-
-    // Update temporary copies of metadata, input- and output definitions to the same state as the resolved graph
-    ORT_RETURN_IF_ERROR_SESSIONID_(SaveModelMetadata(*model_));
+    if (!loading_ort_format) {
+      // add predefined transformers
```

Comment:
DO you want to add some comments here about why we not run optimizer when load ort format

Reply:
Should be covered by my PR (that's now checked in)

---
In reply to: [524842508](https://github.com/microsoft/onnxruntime/pull/5771#discussion_r524842508) [](ancestors = 524842508)

Reply:
To avoid unnecessary conflicts due to the comments, I am not adding any of my own comments.


---
In reply to: [524842508](https://github.com/microsoft/onnxruntime/pull/5771#discussion_r524842508) [](ancestors = 524842508)